linenoise.c

```diff
@@ -1168,7 +1170,11 @@ int linenoiseHistorySave(const char *filename) {
     fp = fopen(filename,"w");
     umask(old_umask);
     if (fp == NULL) return -1;
-    chmod(filename,S_IRUSR|S_IWUSR);
+    if (chmod(filename,S_IRUSR|S_IWUSR) < 0) {
```

Comment:
Unless you want "run a system call on it" to happen before "get a file handle", for some reason